spec/slack-ruby-bot/commands/commands_before_and_after_command_spec.rb

```diff
@@ -0,0 +1,347 @@
+require 'spec_helper'
+
+describe SlackRubyBot::Commands do
+  def app
+    SlackRubyBot::App.new
+  end
+
+  before(:all) do
+    @klass = Class.new(SlackRubyBot::Commands::Base) do
```

Comment:
I'm going to try to figure out `ActiveSupport::Callbacks` to see if that's a suitable replacement first. I'll do this work on another branch. If I get it working and it isn't too ugly, I'll close this pull request and open a new one.